src/components/top-bar/top-bar.stories.tsx

```diff
@@ -0,0 +1,150 @@
+
+          <ElTopBarSecondaryNavContainer>
+            <ElButtonNavIconItem>
+              <Icon icon="star" />
+            </ElButtonNavIconItem>
+            <ElButtonNavIconItem>
+              <Icon icon="star" />
+            </ElButtonNavIconItem>
+            <ElButtonNavIconItem>
+              <Icon icon="star" />
+            </ElButtonNavIconItem>
+          </ElTopBarSecondaryNavContainer>
+
+          <ElTopBarMenuButtonContainer>
+            <ElButtonNavIconItem>
+              <MenuIcon className={elIcon} />
+            </ElButtonNavIconItem>
+          </ElTopBarMenuButtonContainer>
+
+          <ElAvatarButton className={elTopBarAvatar} aria-label="user navigation menu">
+            <ElAvatar data-size="small" data-shape="circle" data-colour="purple">
+              AD
+            </ElAvatar>
+          </ElAvatarButton>
+        </ElTopBar>
+      </nav>
+    )
+  },
+}
+export const ReactUsage: StoryObj<typeof TopBar> = {
+  render: () => {
+    return (
+      <TopBar>
+        <TopBar.MainNavigations
+          style={{
+            containerName: 'main-nav',
+            containerType: 'inline-size',
+          }}
+        >
+          <NavItem href="/">Button 1</NavItem>
+          <NavItem href="/">Button 2</NavItem>
+          <NavItem href="/">Button 3</NavItem>
+          <NavItem href="/">Button 4</NavItem>
+          <ContainerQuery conditions={'(width < 1000px)'} containerName="main-nav">
+            <NavItem href="/">Button 5</NavItem>
+            <NavItem href="/">Button 6</NavItem>
+            <NavItem href="/">Button 7</NavItem>
+          </ContainerQuery>
+          <ContainerQuery not conditions={'(width < 1000px)'} containerName="main-nav">
+            <Menu>
+              <Menu.Trigger>
+                {({ getTriggerProps }) => <NavDropdownButton {...getTriggerProps()}>More</NavDropdownButton>}
+              </Menu.Trigger>
+              <Menu.Popover>
+                <Menu.List>
+                  <Menu.Item href="/">Button 5</Menu.Item>
+                  <Menu.Item href="/">Button 6</Menu.Item>
+                  <Menu.Item href="/">Button 7</Menu.Item>
+                </Menu.List>
+              </Menu.Popover>
+            </Menu>
+          </ContainerQuery>
+        </TopBar.MainNavigations>
+        <TopBar.SearchContainer>
+          <NavIconItem icon={<Icon icon="search" />} aria-label="nav-icon-item-example" />
+          <NavSearchButton />
+        </TopBar.SearchContainer>
+
+        <TopBar.SecondaryNavigations>
+          <NavIconItem aria-label="example" icon={<Icon icon="star" />} />
+          <NavIconItem aria-label="example" icon={<Icon icon="star" />} />
+          <NavIconItem aria-label="example" icon={<Icon icon="star" />} />
+        </TopBar.SecondaryNavigations>
+        <TopBar.MenuButtonContainer>
+          <NavIconItem aria-label="mobile secondary nav trigger" icon={<MenuIcon className={elIcon} />} />
+        </TopBar.MenuButtonContainer>
```

Comment:
with this pattern user have to implement the desktop and mobile behaviour (search icon button, menu button .etc) by themself

Reply:
because of this I think this part is more efficient if we use the props approach instead .e.g `<TopBar.SecondaryNav items={items} />` that also can be applied for `MainNav` and `TopBar.SearchButton`.

Reply:
> I think this part is more efficient if we use the props approach instead .e.g <TopBar.SecondaryNav items={items} />

As per my response on the PR's overview tab, we do not want to move to an interface based on an array of objects. That walks us into a whole lot of downsides; it's much better for us to stick with composing atoms via JSX instead of abstracting atoms into JavaScript objects.